packages/bug-detectors/index.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2023 Code Intelligence GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { registerCommandInjectionBugDetectors } from "./command-injection";
+
+export class BugDetectorError extends Error {}
+
+/**
+ * Registers bug detectors based on the provided list of bug detectors.
+ */
+export async function registerBugDetectors(
+	bugDetectors: string[]
+): Promise<void> {
+	// Keep track of registered bug detectors to avoid registering the same bug detector multiple times.
+	const registeredBugDetectors: string[] = [];
+	for (let i = 0; i < bugDetectors.length; i++) {
+		if (registeredBugDetectors.includes(bugDetectors[i])) {
+			continue;
+		}
+		switch (bugDetectors[i]) {
+			case "commandInjectionSafe":
+				registeredBugDetectors.push(bugDetectors[i]);
+				await registerCommandInjectionBugDetectors(false);
```

Comment:
To be honest I would not worry too much about possible dangerous calls here and simply invoke the original function all the time. Another option could be to add a global flag to suppress calls to the original function, though.

Reply:
The tests are all safe and not allowed to freestyle :man_dancing:, but the example isn't and Murphy's law is a thing.
A global flag seems like a good solution for now, I will add it.